SocialCareCaseViewerApi/V1/Controllers/SocialCareCaseViewerApiController.cs

```diff
@@ -388,19 +388,18 @@ public IActionResult PostWarningNote([FromBody] PostWarningNoteRequest request)
         }
 
         /// <summary>
-        /// Get warning notes by person id
+        /// Get all warning notes created for a specific person
         /// </summary>
-        /// <param name="request"></param>
         /// <response code="200">Success. Returns warning notes related to the specified ID</response>
         /// <response code="404">No warning notes found for the specified ID</response>
-        [ProducesResponseType(typeof(List<WarningNote>), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(ListWarningNotesResponse), StatusCodes.Status200OK)]
```

Comment:
Is there a reason why we want to return `ListWarningNotesResponse` instead of `List<WarningNote>`

Considering `ListWarningNotesResponse` is a class with a single property of type `List<WarningNote>` the class does not seem required.

The different for the frontend is the json with ListWarningNotesResponse would look something like {warningNotes: [{note1}, {note2}]} .

With a List<WarningNote> the response would be more like [{note1}, {note2}].

I've noticed this pattern with a single property where the class does not need to exist elsewhere and in some places has been updated to delete the unrequired class from the codebase.

Reply:
Not sure what the best practice is, but I've seen a lot of APIs return a list in the format of { key: ["item", "item2"] }.
According to our [swagger documentation](https://dr03nduqxh.execute-api.eu-west-2.amazonaws.com/staging/swagger/index.html), there are some endpoints returning responses in this format and some that don't.

I don't think it's a bad format to follow (it looks tidier, in my opinion 😅 )

So maybe this could be something we could discuss as a team? Since we have API documentation containing GET endpoints that don't follow a strict format.

Reply:
The reason I don't like is that the List class already exists that behaves generically with any object type we give it.

It feels "wrong" to make a class called ListX when List<X> is just as easy but doesn't require the creation of a wrapper class.

I agree a quick team discussion on this would be useful so the backend can be consistent.